### PR TITLE
drivers: wifi: remove dependency on NET_SOCKETS_POSIX_NAMES

### LINKED ISSUE #26033: NET_SOCKETS_OFFLOAD conflicts with POSIX_API
**Describe the bug**
When using NET_SOCKETS_OFFLOAD, the symbol NET_SOCKETS_POSIX_NAMES is selected. This symbol conflicts with POSIX_API.

**To Reproduce**
1. Select CONFIG_NET_SOCKETS_OFFLOAD=y
2. Select CONFIG_POSIX_API=y
3. west build
4. See error in configuration conflict

**Expected behavior**
I should be able to use NET_SOCKETS_OFFLOAD and POSIX_API at the same time. NET_SOCKETS_OFFLOAD should not depend on NET_SOCKETS_POSIX_NAMES.

**Impact**
I cannot use POSIX_API when using NET_SOCKETS_OFFLOAD.

**Additional context**
A similar issue was reported in https://github.com/zephyrproject-rtos/zephyr/issues/17353. It sounds like POSIX_API is the "preferred" way of using POSIX BSD sockets in my application. Perhaps NET_SOCKETS_POSIX_NAMES should be deprecated and removed, with breakages fixed where necessary.

### PR DESCRIPTION
**Pull request by aport**
This change removes references to raw POSIX types and functions,
allowing the drivers to build without NET_SOCKETS_POSIX_NAMES.

After this, the dependency between NET_SOCKETS_OFFLOAD and
NET_SOCKETS_POSIX_NAMES can be removed.

See issue #26033 for additional context

Closes #26033

Signed-off-by: Adam Porter <porter.adam@gmail.com>

### CODE REVIEW

**Review by jukkar — APPROVED** [review]

**Review by loicpoulain — APPROVED** [review]

**Comment by pfalcon** [comment]
Please don't merge this without review from @vanti .

**Comment by carlescufi** [comment]
@vanti please take a look

**Review by vanti — APPROVED** [review]
LGTM

**Pull request merged**
